### PR TITLE
Feature/icons

### PR DESCRIPTION
**Pull request by lautimfernandez**
## Motivation
Icon now are used by string path

## Screenshots
This is a UI library, delight us with some stunning images.

## Checks
Are you sure you followed all those steps? In such case, let's say with me "I solemnly declare that ...":
- [ ] I have coded an example inside the Demo App,
- [ ] I have added proper tests,
- [X] I have modified the [Changelog](https://github.com/mercadolibre/fury_andesui-android/blob/develop/CHANGELOG.md) file,
- [ ] I have updated GitHub wiki,
- [ ] I have the explicit approval of one or more members of the UX Team,
- [X] I have updated the version in gradle.properties file.

## Change type
This is easy, let us know what this code is about:
- [ ] This code adds a new component
- [X] This code includes improvements to an existent component
- [ ] This code improves or modifies ONLY the Demo App.

## Check common errors
Whether you are an Android or iOS developer and if you're still there then we'll give you a present:
https://proandroiddev.com/how-to-maximize-androids-ui-reusability-5-common-mistakes-cb2571216a9f

### CODE REVIEW

**Comment by furyGorgoryAdmin** [comment]
### Informative Menu <h2>  ### 

### Use any of these tags in a pull request description or leave a comment ### 
| Tag | Description |
|---|---|
| **[canary]**| Creates a build that can only be deployed with canary strategy. | 
| **[build]** |  Creates version or build from a branch that has been tagged as "no releasable". | 
| **[name:version-name]** |Specify the name of the build. For example: [build] [name:1.2.3-name] or [canary] [name:1.2.3-name].|
| **[major]** | To automatically increase the first position of the version. For example: [build] [major] or [canary] [major].|
| **[minor]** | To automatically increase the second position of the version. For example: [build] [minor] or [canary] [minor]. |
| **[patch]** | To automatically increase the third position of the version. For example: [build] [patch] or [canary] [patch].|  

[For more information about ci-cd](http://docs.furycloud.io/?lang=esp&section=businessassurance)
[For more information about semver](https://semver.org/#summary)

File: components/build.gradle [hunk 1]
```diff
@@ -57,6 +57,7 @@ dependencies {
     implementation "com.android.support:appcompat-v7:$supportLibraryVersion"
     api "com.android.support.constraint:constraint-layout:$constraintLayoutVersion"
     api "com.android.support:cardview-v7:$supportLibraryVersion"
+    implementation 'com.facebook.fresco:fresco:1.13.0'
```
Comment (juanimoli) [inline components/build.gradle]: pasalo a una property en gradle.properties

File: components/src/main/java/com/mercadolibre/android/andesui/button/factory/AndesButtonAttrsParser.kt [hunk 2]
```diff
@@ -12,8 +11,8 @@ import com.mercadolibre.android.andesui.button.size.AndesButtonSize
  */
 internal data class AndesButtonAttrs(val andesButtonHierarchy: AndesButtonHierarchy,
                                      val andesButtonSize: AndesButtonSize,
-                                     val andesButtonLeftIcon: Drawable?,
-                                     val andesButtonRightIcon: Drawable?,
+                                     val andesButtonLeftIcon: String?,
```
Comment (juanimoli) [inline components/src/main/java/com/mercadolibre/android/andesui/button/factory/AndesButtonAttrsParser.kt]: estos estaria bueno cambiar el nombre de la variable para que sea mas descriptivo, onda "andesButtonRightIconDrawableId" o algo asi

File: components/src/main/java/com/mercadolibre/android/andesui/button/factory/AndesButtonConfigurationFactory.kt [hunk 3]
```diff
@@ -156,7 +154,7 @@ internal object AndesButtonConfigurationFactory {
      * @param rightIcon probable icon of the button. Needed because the margins are different if the button has icon or not.
      * @param context needed for accessing dimen resources.
      */
-    private fun resolveMargin(size: AndesButtonSizeInterface, leftIcon: Drawable?, rightIcon: Drawable?, context: Context) = AndesButtonMargin(size, leftIcon, rightIcon, context)
+    private fun resolveMargin(size: AndesButtonSizeInterface, leftIcon: String?, rightIcon: String?, context: Context) = AndesButtonMargin(size, leftIcon, rightIcon, context)
```
Comment (juanimoli) [inline components/src/main/java/com/mercadolibre/android/andesui/button/factory/AndesButtonConfigurationFactory.kt]: idem en todo KDoc/constructor/variable, va a quedar mas descriptivo

File: components/src/main/java/com/mercadolibre/android/andesui/button/size/AndesButtonSizeInterface.kt [hunk 4]
```diff
@@ -139,7 +139,7 @@ internal class AndesLargeButtonSize : AndesButtonSizeInterface {
 
         if (rightIcon != null) {
             val rightBitmapDrawable = buildScaledColoredBitmapDrawable(
-                    rightIcon as BitmapDrawable,
+                    OfflineIconProvider(context).loadIcon(rightIcon) as BitmapDrawable,
```
Comment (juanimoli) [inline components/src/main/java/com/mercadolibre/android/andesui/button/size/AndesButtonSizeInterface.kt]: que onda acá si devuelve null? Lo vamos a handlear de alguna manera?

File: components/src/main/java/com/mercadolibre/android/andesui/icons/IconProvider.kt [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+import android.graphics.drawable.Drawable
+import android.support.v4.content.ContextCompat
+import java.io.FileNotFoundException
+
+
+internal interface IconProvider {
+    fun loadIcon(name: String): Drawable?
+}
+
+internal class OfflineIconProvider(private val context: Context) : IconProvider {
```
Comment (juanimoli) [inline components/src/main/java/com/mercadolibre/android/andesui/icons/IconProvider.kt]: KDoc

File: components/src/main/java/com/mercadolibre/android/andesui/icons/IconProvider.kt [hunk 6]
```diff
@@ -0,0 +1,24 @@
+package com.mercadolibre.android.andesui.icons
+
+
+import android.content.Context
+import android.graphics.drawable.Drawable
+import android.support.v4.content.ContextCompat
+import java.io.FileNotFoundException
+
+
+internal interface IconProvider {
```
Comment (juanimoli) [inline components/src/main/java/com/mercadolibre/android/andesui/icons/IconProvider.kt]: KDoc

File: components/src/main/java/com/mercadolibre/android/andesui/icons/IconProvider.kt [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    fun loadIcon(name: String): Drawable?
```
Comment (juanimoli) [inline components/src/main/java/com/mercadolibre/android/andesui/icons/IconProvider.kt]: KDoc

File: components/src/main/java/com/mercadolibre/android/andesui/message/hierarchy/AndesMessageHierarchyInterface.kt [hunk 8]
```diff
@@ -60,7 +61,7 @@ internal sealed class AndesMessageHierarchyInterface {
      */
     fun dismissableIcon(hierarchy: AndesMessageHierarchyInterface, context: Context) =
             buildColoredBitmapDrawable(
-                    ContextCompat.getDrawable(context, R.drawable.andesui_ui_close_20) as BitmapDrawable,
+                    OfflineIconProvider(context).loadIcon("andesui_ui_close_20") as BitmapDrawable,
```
Comment (juanimoli) [inline components/src/main/java/com/mercadolibre/android/andesui/message/hierarchy/AndesMessageHierarchyInterface.kt]: idem arriba con el null check

File: components/src/test/java/com/mercadolibre/android/andesui/button/AndesButtonSizeTest.kt [hunk 9]
```diff
@@ -68,12 +68,12 @@ class AndesLargeButtonSizeTest {
 
     @Test
     fun `Large button left icon`() {
-        assertEquals(andesLargeButtonSize.iconConfig(hierarchy, icon, null, context), IconConfig(icon, null))
+        assertEquals(andesLargeButtonSize.iconConfig(hierarchy, "andesui_icon", null, context), IconConfig(icon, null))
```
Comment (juanimoli) [inline components/src/test/java/com/mercadolibre/android/andesui/button/AndesButtonSizeTest.kt]: pasalo a constante asi no lo repetis en todos los tests :P

File: demoapp/build.gradle [hunk 10]
```diff
@@ -74,6 +75,7 @@ dependencies {
     implementation "com.android.support.constraint:constraint-layout:$constraintLayoutVersion"
     implementation "com.android.support:appcompat-v7:$supportLibraryVersion"
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk7:$kotlinVersion"
+    implementation 'com.facebook.fresco:fresco:1.13.0'
```
Comment (juanimoli) [inline demoapp/build.gradle]: idem gradle properties

File: demoapp/src/main/java/com/mercadolibre/android/andesui/demoapp/MainApplication.kt [hunk 11]
```diff
@@ -11,10 +17,21 @@ class MainApplication : Application() {
...
         // No need for productFlavors, as proguard will remove all multidex related code in non-debug builds.
         if (BuildConfig.BUILD_TYPE.contentEquals("debug")) {
             MultiDex.install(this)
         }
     }
+
+    override fun onCreate() {
+        super.onCreate()
+
+        val requestListeners = setOf<RequestListener>(RequestLoggingListener())
```
Comment (juanimoli) [inline demoapp/src/main/java/com/mercadolibre/android/andesui/demoapp/MainApplication.kt]: esto lo vamos a tener que agregar en la app tambien?

Comment (lautimfernandez) [reply demoapp/src/main/java/com/mercadolibre/android/andesui/demoapp/MainApplication.kt]: ya esta en la app el import de fresco, no es necesario

File: components/src/main/java/com/mercadolibre/android/andesui/icons/IconProvider.kt [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+ */
+internal class OfflineIconProvider(private val context: Context) : IconProvider {
+
+    override fun loadIcon(name: String): Drawable? {
+        val resId = context.resources.getIdentifier(name, "drawable", context.packageName)
+        return try {
+            ContextCompat.getDrawable(context, resId)
+        } catch (e: FileNotFoundException) {
+            null
+        }
```
Comment (jiurchuk) [inline components/src/main/java/com/mercadolibre/android/andesui/icons/IconProvider.kt]:
```suggestion
        } catch (e: FileNotFoundException) {
            Log.e("Andes UI", "File $name was not found.")
            null
        }
```

**Pull request merged**